plan/relations.go

```diff
@@ -267,6 +302,13 @@ func (v *VirtualTableReadRel) CopyWithExpressionRewrite(rewriteFunc RewriteFunc,
 	return &vtr, nil
 }
 
+func (v *VirtualTableReadRel) Remap(mapping ...int32) (Rel, error) {
+	newMapping, err := RemapHelper(v, mapping)
```

Comment:
If I were you, I'd introduce a non exported set mapping function on relation and then this code could be factored out to a single invocation.


something like 
```
RemapHelper(r Rel, newMapping) (Rel, error) {
  newRel := r.Copy(v.GetInputs())
  err := r.setMapping(newMapping)
  return newRel, err
}
```

Reply:
Adding the unexported method to the Rel interface was the trick I needed to get that to work.  Thanks!